benchmark/src/main/java/org/neo4j/graphalgo/bench/MSBFSBenchmark.java

```diff
@@ -0,0 +1,129 @@
+package org.neo4j.graphalgo.bench;
+
+import algo.Pools;
+import org.neo4j.graphalgo.api.IdMapping;
+import org.neo4j.graphalgo.api.RelationshipConsumer;
+import org.neo4j.graphalgo.api.RelationshipCursor;
+import org.neo4j.graphalgo.api.RelationshipIterator;
+import org.neo4j.graphalgo.impl.msbfs.BfsConsumer;
+import org.neo4j.graphalgo.impl.msbfs.MultiSourceBFS;
+import org.neo4j.graphdb.Direction;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+@Threads(1)
+@Fork(value = 1, jvmArgs = {"-Xms4G", "-Xmx4G"})
```

Comment:
suggest running with `-XX:-UseSuperWord` (prevent JVM from using vector/SIMD operations) & `-XX:+UseSuperWord` (allow JVM to use vector/SIMD operations)

this will tell you if your code is written in a way that SIMD optimizations can be made (that is a major contribution of this paper/algorithm)

regarding SIMD & Java: http://prestodb.rocks/code/simd/